packages/pass-style/src/types.d.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable no-use-before-define */
+import { PASS_STYLE } from './passStyle-helpers.js';
+
+export type PrimitiveStyle =
+  | 'undefined'
+  | 'null'
+  | 'boolean'
+  | 'number'
+  | 'bigint'
+  | 'string'
+  | 'symbol';
+export type ContainerStyle = 'copyRecord' | 'copyArray' | 'tagged';
+export type PassStyle =
+  | PrimitiveStyle
+  | ContainerStyle
+  | 'remotable'
+  | 'error'
+  | 'promise';
+export type PassStyled<S extends string> = {
+  [PASS_STYLE]: S;
+};
+export type ExtractStyle<P extends PassStyled<any>> = P[typeof PASS_STYLE];
+export type PassByCopy =
+  | import('type-fest').Primitive
+  | Error
+  | any[]
+  | CopyRecord
+  | CopyTagged;
+export type PassByRef =
+  | RemotableObject
+  | Promise<RemotableObject>
+  | Promise<PassByCopy>;
```

Comment:
I was going to suggest

```suggestion
export type PassByRef =
  | RemotableObject
  | Promise<Passable>;
```
but then I realized that your's correctly excludes `Promise<Promise>`, which is great! No change suggested.

Reply:
You could likely do `Promise<Awaited<Passable>>`